arxiv_public_data/internal_citations.py

```diff
@@ -16,21 +16,30 @@
 RE_FLEX = re.compile(REGEX_ARXIV_FLEXIBLE)
 RE_OLDNAME_SPLIT = re.compile(r"([a-z\-]+)(\d+)")
 
+
 def path_to_id(path):
     """ Convert filepath name of ArXiv file to ArXiv ID """
     name = os.path.splitext(os.path.basename(path))[0]
     if '.' in name:  # new  ID
-        return name 
+        return name
     split = [a for a in RE_OLDNAME_SPLIT.split(name) if a]
     return "/".join(split)
 
+
 def all_articles(directory=DIR_FULLTEXT):
     """ Find all *.txt files in directory """
     out = []
+    # make sure the path is absolute for os walk
+    if directory[0] == '~':
+        directory = os.path.expanduser(directory)
+    elif directory != '/':
+        directory = os.path.abspath(directory)
```

Comment:
Why is the logic necessary? I believe you can just always apply `abspath` and it should work for both cases.

Reply:
`abspath` doesn't work for something like `~/example`, it returns  `cwd/~/example` instead of expand `~` to $HOME

Reply:
You're right! I think I meant that `expanduser` has no effect if there is no `~`, so you may as well always apply it.

Reply:
I applied both at once, see new [commit ](https://github.com/mattbierbaum/arxiv-public-datasets/pull/14/commits/6fda3f780d5bd670e94bdb57be2d97ac6271b86e)